### PR TITLE
MBEDTLS config-coap should use standard max content length

### PR DESCRIPTION
**Pull request by mike-scott**
The CoAP samples use an MBEDTLS config "config-coap.h" which could be
re-used by the LwM2M sample, except that most servers use a larger
maximum content length setting of 1500 bytes.

Let's add a CONFIG to set this for users of the CoAP lib and then adjust the coap samples to use the smaller value.

### CODE REVIEW

**Comment by mike-scott** [comment]
NOTE: This fixes DTLS on LwM2M and is required for https://github.com/zephyrproject-rtos/zephyr/pull/5267

**Review by rbtchc — APPROVED** [review]
LGTM

**Review by pfalcon — APPROVED** [review]
This is a rare case when I'd suggest to squash properly separated patches into one for being even more proper ;-).

**Review by rveerama1 — APPROVED** [review]
LGTM

**Comment by mike-scott** [comment]
@pfalcon Were you ok as is?  Or rather I squashed these 2 together?  @jukkar How do you feel about squashing?

**Comment by pfalcon** [comment]
> Were you ok as is? Or rather I squashed these 2 together?

As I approved it, I was. But I think it would be better to squash them. Just consider that 1st patch changes the behavior (setting), and 2nd restores it back. Why this flip-flopping, as it's cheap to do it atomically here. Of course, you may your own arguments, so up to you.

> @jukkar How do you feel about squashing?

@jukkar is on vacation.

File: subsys/net/lib/coap/Kconfig [hunk 1]
```diff
@@ -62,6 +62,15 @@ config COAP_EXTENDED_OPTIONS_LEN_VALUE
 	COAP_EXTENDED_OPTIONS_LEN is enabled. Define the value according to
 	user requirement.
 
+config COAP_MBEDTLS_SSL_MAX_CONTENT_LEN
+	int "CoAP MBEDTLS maximum content length value"
+	default 1500
```
Comment (lpereira) [inline subsys/net/lib/coap/Kconfig]: Does this have any minimum size? If so, please add a `range` statement.  Otherwise, this change LGTM.

**Comment by mike-scott** [comment]
Posted V2:
- Squashed the 2 commits into an atomic change per @pfalcon suggestion
- Added a range from 256 to 16k per @lpereira suggestion

**Review by lpereira — APPROVED** [review]

**Comment by mike-scott** [comment]
Rebased on today's master due to conflict.

**Pull request merged**